app/auth/context/AuthContext.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { createContext, useContext, PropsWithChildren } from 'react';
+import { useStorageState } from '../hooks/useStorageState';
+import { router } from 'expo-router';
+
+const API = process.env.EXPO_PUBLIC_API_URL || 'http://localhost:4000';
+
+type AuthDataType = {
+  email: string;
+  accessToken: string;
+};
+
+type AuthContextType = {
+  user: AuthDataType | null;
+  signIn: (email: string, password: string) => Promise<void>;
+  signUp: (email: string, username: string, password: string) => Promise<void>;
+  signOut: () => Promise<void>;
+  isLoading: boolean;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export function useAuth() {
+  const value = useContext(AuthContext);
+  if (!value) {
+    throw new Error('useAuth must be wrapped in an <AuthProvider />');
+  }
+
+  return {
+    ...value,
+  };
+}
+
+export function AuthProvider({ children }: PropsWithChildren) {
```

Comment:
I see `fat arrow` and `function` here, can we stick to one of them? preferably `fat arrow`.

Reply:
I changed this one but it's still used often throughout the project, expo doesn't run with fat arrow exports on navigation (tab) pages. We might want to just stick to using `export default function` throughout the project since it doesn't support the other, let me know.